src/umpire/resource/MemoryResourceRegistry.cpp

```diff
@@ -108,27 +108,27 @@ MemoryResourceRegistry::MemoryResourceRegistry() : m_allocator_factories()
     int device_count{0};
     auto error = ::cudaGetDeviceCount(&device_count);
     if (error != cudaSuccess) {
-      UMPIRE_ERROR("Umpire compiled with CUDA support but no GPUs detected!");
-    }
-
-    registerMemoryResource(util::make_unique<resource::CudaDeviceResourceFactory>());
-    m_resource_names.push_back("DEVICE");
-
-    for (int device = 0; device < device_count; ++device) {
-      std::string name{"DEVICE::" + std::to_string(device)};
-      m_resource_names.push_back(name);
-    }
+      UMPIRE_LOG(Warning, "Umpire compiled with CUDA support but no GPUs detected!");
```

Comment:
Is there a specific error code that can be checked for device existence?  I assume that in theory, other error codes code be returned?